controller/src/controller/models/domain/imagefilterpolicy.py

```diff
@@ -0,0 +1,133 @@
+"""Policy for selecting images based on filter criteria."""
+
+from typing import Annotated
+
+from pydantic import BaseModel, BeforeValidator, ConfigDict, Field
+from pydantic.alias_generators import to_camel
+from safir.pydantic import HumanTimedelta
+from semver import Version
+
+__all__ = ["ImageFilterPolicy", "RSPImageFilterPolicy"]
+
+
+class ImageFilterPolicy(BaseModel):
+    """Policy for images to display within a given category.
+
+    All specified policies will be applied.  For instance, if the policy
+    specifies both age and cutoff version, then an image will have to be
+    newer than the specified age, and also have a version greater than or
+    equal to the cutoff, in order to be displayed.
+
+    If no policies are specified, no filtering will be performed and all
+    images of that category will be displayed.
+    """
+
+    model_config = ConfigDict(
+        alias_generator=to_camel,
+        extra="forbid",
+        populate_by_name=True,
+        arbitrary_types_allowed=True,
+    )
+
+    number: Annotated[
+        int | None,
+        Field(
+            title="Number",
+            description="Number of images to display.",
+            ge=0,
+        ),
+    ] = None
+
+    age: Annotated[
+        HumanTimedelta | None,
+        Field(
+            title="Age",
+            description="Maximum age of image to display.",
+        ),
+    ] = None
+
+    cutoff_version: Annotated[
+        Version | None,
+        BeforeValidator(lambda v: v if not isinstance(v, str) else Version(v)),
+        Field(
+            title="Cutoff Version",
+            description=(
+                "Minimum version of image to display."
+                " This does not apply to unparseable tags or to"
+                " experimental tags not derived from a parseable tag."
+            ),
+        ),
+    ] = None
+
+
+class RSPImageFilterPolicy(BaseModel):
+    """Configuration for display of RSP images."""
+
+    model_config = ConfigDict(
+        alias_generator=to_camel, extra="forbid", populate_by_name=True
+    )
+
+    alias: Annotated[
+        ImageFilterPolicy,
+        Field(
+            title="Release",
+            description="Policy for releases to display.",
+            default_factory=ImageFilterPolicy,
+        ),
+    ]
```

Comment:
I don't think filtering aliases this way really makes sense. I think we want to include the alias iff the underlying tag is included.

Reply:
While I agree with you that filtering aliases doesn't make much sense, if we don't special-case the alias category, the implementation of the filter becomes (to my mind) quite a bit cleaner.

https://github.com/lsst-sqre/nublado/blob/05132eff2fb7901b39609256e68340c9702e1b33/controller/src/controller/models/domain/rsptag.py#L636-L655

Reply:
If we do special case that, there's an extra test that feels awkward inside the loop.  If you care a lot I'll put it back that way, but that's how I wrote it the first time and it didn't feel as nice as just letting the alias category be handled like all the others.

Reply:
If you'd rather not implement that filtering for now, I think that's fine, but in that case I think we should just not filter alias tags at all. That avoids the problem of what the right semantic solution is, and we can always add filtering for alias tags later.